### DISCUSSION TITLE
LoRaWAN Sensor Node | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#1933] [Q&A]
# LoRaWAN Sensor Node | Seeed Studio Wiki

Lorawan sensor node based on Wio SX1262 with XIAO esp32s3 module

https://wiki.seeedstudio.com/wio_sx1262_xiao_esp32s3_for_lora_sensor_node/

**Comment by KARTHIKEYAN4** [comment]
Hi , in the above example code how to switch Class A & Class C devices and Downlink source code is not available

**Reply by KARTHIKEYAN4** [reply]
Hi , in the above example code how to switch Class A & Class C devices and Downlink source code is not available

**Reply by cure123123** [reply]
Sorry for the late reply. Regarding the first question, it is recommended that you refer to the GitHub of Radiolib and adjust the code to configure it as the corresponding Class A or C. As for the second question, could you please provide me with more information? We have verified that it works, which will facilitate me to provide technical support.

**Reply by pulkitjn97** [reply]
Refer to this example from RadioLib:
https://github.com/jgromes/RadioLib/tree/master/examples/LoRaWAN/LoRaWAN_Starter

**Comment by michapr** [comment]
Would be nice to see a deep sleeping sample code.
I cannot get low current - in any case only the SX1262 module take about additional 600uA while sleeping - so a battery powered device is not possible in this time...

**Reply by Zeke5579** [reply]
We not sure how you use it, but you can refer here to try low power sleep mode: https://github.com/Lora-net  & 
 https://github.com/Lora-net/SWSD003

**Reply by lhanneus** [reply]
Yes LoRaWan with deepsleep example is needed ! And top would be wake up on a time basis AND on an external trigger (an on/off switch for example).

**Comment by ranga1253** [comment]
The arduino code isnt working on my node but i can not find firmware files to edit and upload manually for lorawan sensor node operation. Also why does xaio esp32 s3 come advertised as lorawan but then have firmware flashed for meshtastic?

**Reply by michapr** [reply]
What code you are using? And what is not working? Can you post the code and the error messages?

**Reply by ranga1253** [reply]
I was able to find the proper firmware but don't exactly how to see if this was all done properly. 

I don't see anything coming up on the screen on the breakout board, but in terminal i get 
```
Join failed - RADIOLIB_LORAWAN_NO_DOWNLINK (-1116)
```
so obviously seems like an error since it doesn't see a gateway matching its config. Is there a way to see if this is set up properly so i can set up a gateway with a known good edge sensor? Is there a way to read sensor data via serial?

**Reply by michapr** [reply]
I think it is more a configuration problem.
In RadioLib (I think you are using this lib...) documentation you can see:
```
RADIOLIB_ERR_NO_JOIN_ACCEPT   (-1116)
No JoinAccept was received - check your keys, or otherwise likely a range issue!
```
(https://jgromes.github.io/RadioLib/group__status__codes.html)

I'm using the XIAO S3 device with a CO2-sensor and it is working fine with TTN (and also with local ChirpStack server).
Maybe it's a good idea to check the RadioLib forum for configuration notes?

BTW: Serial output (USB terminal) is working fine in my case.

**Comment by pradeeplogu0** [comment]
Hi Team, I'm trying to connect the kit to TTN but it can't able to connect to the WiFi, this is the response I'm getting from the kit.

I (299) esp_image: segment 4: paddr=00118bb8 vaddr=4037bbI (4799) wifi:state: run -> init (f00)
I (4809) wifi:new:<6,0>, old:<6,1>, ap:<255,255>, sta:<6,1>, prof:1
I (4809) wifi: WIFI_EVENT_STA_DISCONNECTED
I (4809) wifi: Disconnected from Sellamuthu for reason 15
I (4819) wifi: retry to connect to the AP (1)
I (6509) wifi: WIFI_EVENT_STA_DISCONNECTED
I (6509) wifi: Disconnected from Sellamuthu for reason 205
I (6509) wifi: retry to connect to the AP (2)
I (6519) wifi:new:<6,1>, old:<6,0>, ap:<255,255>, sta:<6,1>, prof:1
I (6519) wifi:state: init -> auth (b0)
I (6529) wifi:state: auth -> assoc (0)
I (6539) wifi:state: assoc -> run (10)
I (9539) wifi:state: run -> init (f00)
I (9539) wifi:new:<6,0>, old:<6,1>, ap:<255,255>, sta:<6,1>, prof:1
I (9539) wifi: WIFI_EVENT_STA_DISCONNECTED
I (9539) wifi: Disconnected from Sellamuthu for reason 15
I (9549) wifi: retry to connect to the AP (3)
I (11239) wifi: WIFI_EVENT_STA_DISCONNECTED
I (11239) wifi: Disconnected from Sellamuthu for reason 205
I (11239) wifi: retry to connect to the AP (4)
I (11249) wifi:new:<6,1>, old:<6,0>, ap:<255,255>, sta:<6,1>, prof:1
I (11249) wifi:state: init -> auth (b0)
I (11259) wifi:state: auth -> assoc (0)
I (11269) wifi:state: assoc -> run (10)
I (14259) wifi:state: run -> init (f00)
I (14269) wifi:new:<6,0>, old:<6,1>, ap:<255,255>, sta:<6,1>, prof:1
I (14269) wifi: WIFI_EVENT_STA_DISCONNECTED
I (14269) wifi: Disconnected from Sellamuthu for reason 15
I (14279) wifi: retry to connect to the AP (5)
I (15969) wifi: WIFI_EVENT_STA_DISCONNECTED
I (15969) wifi: Disconnected from Sellamuthu for reason 205
I (15969) wifi: retry to connect to the AP (6)
I (15979) wifi:new:<6,1>, old:<6,0>, ap:<255,255>, sta:<6,1>, prof:1
I (15979) wifi:state: init -> auth (b0)
I (15989) wifi:state: auth -> assoc (0)
I (15999) wifi:state: assoc -> run (10)
I (18979) wifi:state: run -> init (f00)
I (18989) wifi:new:<6,0>, old:<6,1>, ap:<255,255>, sta:<6,1>, prof:1
I (18989) wifi: WIFI_EVENT_STA_DISCONNECTED
I (18989) wifi: Disconnected from Sellamuthu for reason 15
I (18999) wifi: retry to connect to the AP (7)
I (20689) wifi: WIFI_EVENT_STA_DISCONNECTED
I (20689) wifi: Disconnected from Sellamuthu for reason 205
I (20689) wifi: retry to connect to the AP (8)
I (20699) wifi:new:<6,1>, old:<6,0>, ap:<255,255>, sta:<6,1>, prof:1
I (20699) wifi:state: init -> auth (b0)
I (20709) wifi:state: auth -> assoc (0)
I (20709) wifi:state: assoc -> run (10)
I (23699) wifi:state: run -> init (f00)
I (23709) wifi:new:<6,0>, old:<6,1>, ap:<255,255>, sta:<6,1>, prof:1
I (23709) wifi: WIFI_EVENT_STA_DISCONNECTED
I (23709) wifi: Disconnected from Sellamuthu for reason 15
I (23719) wifi: retry to connect to the AP (9)
I (25409) wifi: WIFI_EVENT_STA_DISCONNECTED
I (25409) wifi: Disconnected from Sellamuthu for reason 205
I (25409) wifi: retry to connect to the AP (10)
I (25419) wifi:new:<6,1>, old:<6,0>, ap:<255,255>, sta:<6,1>, prof:1
I (25419) wifi:state: init -> auth (b0)
I (25429) wifi:state: auth -> assoc (0)
I (25429) wifi:state: assoc -> run (10)
I (28419) wifi:state: run -> init (f00)
I (28429) wifi:new:<6,0>, old:<6,1>, ap:<255,255>, sta:<6,1>, prof:1
I (28429) wifi: WIFI_EVENT_STA_DISCONNECTED
I (28429) wifi: Disconnected from Sellamuthu for reason 15
E (28429) wifi: Failed to connect to AP
E (28439) main: ERROR: failed to initialize WiFi station

**Reply by Jessie219-web** [reply]
make sure you have connected the wifi antenna, and your wifi should be 2.4GHz

**Comment by jackwilliams2480** [comment]
Would have liked a working example where, since you are using for implementation inside TTN, the base code followed the fair use policy. The code also cannot handle the node restarting without removing and readding the device inside the console due to DevNonce errors, otherwise very helpful.

**Reply by Jessie219-web** [reply]
we will consider adding this example later. thanks for your feedback.
